src/features/core/metrics/errorReporting/providers/Bugsnag.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-disable no-underscore-dangle */
+import ScriptLoader from '../../../../../utils/ScriptLoader';
+import { configToIntNames } from '../../../../../utils/config_to_integration_names';
+import {
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
+  MAX_WAIT_FOR_INTEGRATION_LOAD,
+} from '../../../../../utils/constants';
+import { get } from '../../../../../utils/utils';
+
+// Using the Bugsnug integration version to avoid version issues
+const BUGSNUG_CDN_URL = 'https://d2wy8f7a9ursnm.cloudfront.net/v6/bugsnag.min.js';
+const BUGSNUG_VALID_MAJOR_VERSION = '6';
+const ERROR_REPORT_PROVIDER_NAME_BUGSNUG = 'rs-bugsnug';
+const BUGSNUG_LIB_INSTANCE_GLOBAL_KEY_NAME = 'bugsnag'; // For version 7 this is 'Bugsnug'
+const GLOBAL_LIBRARY_OBJECT_NAMES = [
+  ERROR_REPORT_PROVIDER_NAME_BUGSNUG,
+  'Bugsnug',
+  BUGSNUG_LIB_INSTANCE_GLOBAL_KEY_NAME,
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
+];
+
+// This API key token is parsed in the CI pipeline
+const API_KEY = '__RS_BUGSNAG_API_KEY__';
+
+// Errors only from Below SDKs are allowed to reach Bugsnag
+const SDK_FILE_NAMES = [
+  'rudder-analytics.min.js',
+  'rudder-analytics-staging.min.js',
+  'rudder-analytics.js',
+  ...Object.keys(configToIntNames).map((intgName) => `${configToIntNames[intgName]}.min.js`),
+  ...Object.keys(configToIntNames).map(
+    (intgName) => `${configToIntNames[intgName]}-staging.min.js`,
+  ),
+  ...Object.keys(configToIntNames).map((intgName) => `${configToIntNames[intgName]}.js`),
+];
+const BUGSNAG_ERROR_FILE_ORIGIN_KEY = 'errors.0.stacktrace.0.file';
+
+const getReleaseStage = () => {
+  const host = window.location.hostname;
+  const devHosts = ['www.rs-unit-test-host.com', 'localhost', '127.0.0.1', '[::1]'];
+
+  return host && devHosts.includes(host) ? 'development' : '__RS_BUGSNUG_RELEASE_STAGE__';
+};
+
+const isValidVersion = (globalLibInstance) => {
+  // For version 7
```

Comment:
May be some reference docs help here

Reply:
reference docs?

Reply:
adding links to docs can help, I thought.

Reply:
Bugsnag docs? sure. However these values I found them using debugger in browser after instantiation of the Bugsnag SDK

Reply:
https://docs.bugsnag.com/platforms/javascript/